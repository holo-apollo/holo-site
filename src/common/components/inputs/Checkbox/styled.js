// @flow
import styled, { css } from 'styled-components';

import palette from 'common/palette';
import { getEnv } from 'helpers/misc';

const staticRoot = getEnv('STATIC_ROOT') || '';

const getLabelStyles = ({ error }: { error: boolean }) => css`
  color: ${error ? palette.red : palette.grey};
`;

export const StyledLabel = styled.label`
  display: flex;
  align-items: center;
  cursor: pointer;
  font-size: 12px;
  text-align: left;

  ${getLabelStyles};
`;

const getCheckboxStyles = ({ size }) => css`
  width: ${size}px;
  height: ${size}px;
  min-width: ${size}px;
  max-width: ${size}px;

  &:checked {
    background-size: ${(size * 2) / 3}px ${(size * 2) / 3}px;
  }
`;

export const StyledCheckbox = styled.input`
  appearance: none;
  box-sizing: content-box;
  border: solid 1px ${palette.lightGrey};
  margin: 0 8px 0 0;
  padding: 0;
  cursor: pointer;

  &:focus,
  &:active {
    outline: none;
  }

  &:checked {
    background-image: url(${`${staticRoot}/img/checkbox.svg`});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 18px 13px;
    background-color: ${palette.white};
  }

  ${getCheckboxStyles};
`;
