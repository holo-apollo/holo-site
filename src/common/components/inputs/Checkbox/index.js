// @flow
import React from 'react';

import FieldWithError from '../FieldWithError';
import { StyledLabel, StyledCheckbox } from './styled';

type Props = {
  name: string,
  label: string,
  errorText?: string,
  size: number,
};

const Checkbox = ({ name, label, errorText, size, ...rest }: Props) => (
  <FieldWithError errorText={errorText}>
    <StyledLabel error={Boolean(errorText)}>
      <StyledCheckbox size={size} type="checkbox" name={name} {...rest} />
      {label}
    </StyledLabel>
  </FieldWithError>
);

Checkbox.defaultProps = {
  size: 20,
};

export default Checkbox;
