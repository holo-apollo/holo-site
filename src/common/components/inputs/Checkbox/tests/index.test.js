import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from '../index';

describe('Checkbox', function() {
  it('should match snapshot', function() {
    const tree = renderer
      .create(<Checkbox name="name" label="Label" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
