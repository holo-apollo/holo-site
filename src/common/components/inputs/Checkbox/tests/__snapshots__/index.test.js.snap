// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<label
  className="styled__StyledLabel-sc-93nche-0 iATPBy"
>
  <input
    className="styled__StyledCheckbox-sc-93nche-1 jwdXCo"
    name="name"
    size={20}
    type="checkbox"
  />
  Label
</label>
`;
